mentii/mentii
# View class details

Users can now see a class details page with will display:
- class name
- class description
- class department (optional)
- class section (optional)
- list of class's activities
- If the user is the teacher of the class:
   - Teacher View Containing:
     - List of students in class

**Significant Changes**
1. Added getClass rest endpoint to app.py
2. Added getClass controller method to class_ctrl.py, which returns class data, only including students and isTeacher attributes if the user is the class's teacher
3. Added getClassByCode method to class_ctrl.py, which handles the DB interactions for getClass
4. Added tests for getClass and getClassByCode  to test_class_ctrl.py (using Mock to avoid DB interactions)
5. Created an activity model
6. Created an activity list component
7.  Created html for activity list
8. Added new components to app.module.ts
9. Modified class model to include an array of ActivityModel and an array of student strings
10. Added getClass method to class service to hit getClass rest endpoint
11. Modified the class detail component to contain and represent a class model, which it populates using the  class service getClass method. This call also provides data to activate the Teacher View if the user is the class's teacher
12. Updated detail html for above change
13. Style fix to dashboard html
14. Created user list component
15. Created html for user list component
16. Updated version

**How to Test**
1. Pull latest is LocalDB master
2. make
3. Call ./putItem.sh in LocalDB/users
3. Call ./putItem.sh in LocalDB/classes
4. Login as me
6. Go to the public class list and click on the mock class
7.  You should now see its details view, including activities, but you should not see the teacher view on the right
7. Create a new class
8. In your teaching list click the view class details button for your new class
9. You should now see its details view, with no activities, but you should now see the teacher view on the right with no students
10. Go back to the public class list and join your class
11. Go back to the your class's detail view and you should see my email in the students list in the teacher view
